### PR TITLE
feat(cockroachdb): Init and display provider in CLI output

### PR DESCRIPTION
**Pull request by Jolg42**
Related https://github.com/prisma/prisma/issues/10808

### CODE REVIEW

**Comment by SoheilAdmin** [comment]
#[]()

**Comment by Jolg42** [comment]
Looks like this with next commit
Given this schema
```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["cockroachdb"]
}

datasource db {
  provider = "cockroachdb"
  url      = "postgres://user:password@free-tier.gcp-us-central1.cockroachlabs.cloud:26257/dbname-5198.defaultdb?sslmode=verify-full&sslrootcert=$HOME/.postgresql/root.crt&options=--cluster%3Dintrospection-ci-5134"
}
```

>Datasource "db": CockroachDB database "introspection-ci-5134.defaultdb", schema "public" at "free-tier.gcp-us-central1.cockroachlabs.cloud:26257"

File: packages/migrate/src/__tests__/MigrateResolve.test.ts [hunk 1]
```diff
@@ -212,28 +167,43 @@ The migration does_not_exist could not be found. Please make sure that the migra
...
+    expect(ctx.mocked['console.info'].mock.calls.join('\n')).toMatchInlineSnapshot(`
       Prisma schema loaded from prisma/invalid-url.prisma
       Datasource "my_db": PostgreSQL database "mydb", schema "public" at "doesnotexist:5432"
     `)
     expect(ctx.mocked['console.log'].mock.calls).toMatchSnapshot()
     expect(ctx.mocked['console.error'].mock.calls).toMatchSnapshot()
   })
 })
+
+describe('cockroachdb', () => {
```
Comment (aqrln) [inline packages/migrate/src/__tests__/MigrateResolve.test.ts:190]:
```suggestion
const describeIf = (condition: boolean) => (condition ? describe : describe.skip)

describeIf(!process.env.TEST_SKIP_COCKROACHDB)('cockroachdb', () => {
```

Comment (Jolg42) [reply packages/migrate/src/__tests__/MigrateResolve.test.ts:190]: Why do we need this?

Comment (aqrln) [reply packages/migrate/src/__tests__/MigrateResolve.test.ts:190]: CockroachDB isn't installed on Windows and macOS CI

**Review by millsp — APPROVED** [review]

File: packages/migrate/src/utils/ensureDatabaseExists.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import prompt from 'prompts'
 import type execa from 'execa'
 
 export type MigrateAction = 'create' | 'apply' | 'unapply' | 'dev' | 'push'
-export type DbType = 'MySQL' | 'PostgreSQL' | 'SQLite' | 'SQL Server'
+export type DbType = 'MySQL' | 'PostgreSQL' | 'SQLite' | 'SQL Server' | 'CockroachDB'
```
Comment (millsp) [inline packages/migrate/src/utils/ensureDatabaseExists.ts:10]: Unrelated: we don't need MongoDB here?

Comment (Jolg42) [reply packages/migrate/src/utils/ensureDatabaseExists.ts:10]: Since `db push` works with MongoDB, indeed, I'll have a look later at this!

**Pull request merged**